src/statsbase.jl

```diff
@@ -0,0 +1,186 @@
+using StatsBase
+
+
+# Support some of the weighted statistics function in StatsBase
+# NOTES:
+# - Ambiguity errors are still possible for weights with overly specific methods (e.g., UnitWeights)
+# - Ideally, when the weighted statistics is moved to Statistics.jl we can remove this entire file.
+#   https://github.com/JuliaLang/Statistics.jl/pull/2
+function Statistics.mean(A::KeyedArray, wv::AbstractWeights; dims=:, kwargs...)
+    dims === Colon() && return mean(parent(A), wv; kwargs...)
+    numerical_dims = AxisKeys.hasnames(A) ? NamedDims.dim(dimnames(A), dims) : dims
+    data = mean(parent(A), wv; dims=numerical_dims, kwargs...)
+    new_keys = ntuple(d -> d in numerical_dims ? Base.OneTo(1) : axiskeys(A,d), ndims(A))
+    return KeyedArray(data, map(copy, new_keys))#, copy(A.meta))
+end
+
+function Statistics.mean(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return mean(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+# var and std are separate cause they don't use the dims keyword and we need to set corrected=true
+for fun in [:var, :std]
+    @eval function Statistics.$fun(A::KeyedArray, wv::AbstractWeights; dims=:, corrected=true, kwargs...)
+        dims === Colon() && return $fun(parent(A), wv; kwargs...)
+        numerical_dims = AxisKeys.hasnames(A) ? NamedDims.dim(dimnames(A), dims) : dims
+        data = $fun(parent(A), wv, numerical_dims; corrected=corrected, kwargs...)
+        new_keys = ntuple(d -> d in numerical_dims ? Base.OneTo(1) : axiskeys(A,d), ndims(A))
+        return KeyedArray(data, map(copy, new_keys))#, copy(A.meta))
+    end
+
+    @eval function Statistics.$fun(A::KeyedArray, wv::KeyedVector; kwargs...)
+        return $fun(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+    end
+end
+
+function Statistics.cov(A::KeyedMatrix, wv::AbstractWeights; dims=1, corrected=true, kwargs...)
+    # A little awkward, but the weighted `cov` method from statsbase only works
+    # on dense matrices, so we need to unwrap this twice and completely rebuild the
+    # array
+    if AxisKeys.hasnames(A)
+        numerical_dim = NamedDims.dim(dimnames(A), dims)
+        data = cov(parent(parent(A)), wv, numerical_dim; corrected=corrected, kwargs...)
+        rem_dim = first(setdiff((1, 2), numerical_dim))
```

Comment:
I think these only accept `dims::Int`. In which case this can just be `3 - numerical_dim`.